cmd/kubeadm/app/cmd/upgrade/plan_test.go

```diff
@@ -330,19 +345,71 @@ Note: Before you can perform this upgrade, you have to update kubeadm to v1.9.0-
 
 _____________________________________________________________________
 
+`),
+		},
+		{
+			name: "external etcd upgrade available",
+			upgrades: []upgrade.Upgrade{
+				{
+					Description: "version in the v1.9 series",
+					Before: upgrade.ClusterState{
+						KubeVersion: "v1.9.2",
+						KubeletVersions: map[string]uint16{
+							"v1.9.2": 1,
+						},
+						KubeadmVersion: "v1.9.2",
+						DNSVersion:     "1.14.5",
+						EtcdVersion:    "3.0.17",
+					},
+					After: upgrade.ClusterState{
+						KubeVersion:    "v1.9.3",
+						KubeadmVersion: "v1.9.3",
+						DNSVersion:     "1.14.8",
+						EtcdVersion:    "3.1.12",
+					},
+				},
+			},
+			externalEtcd: true,
+			expectedBytes: []byte(`External components that should be upgraded manually before you upgrade the control plane with 'kubeadm upgrade apply':
```

Comment:
these byte comparisons without helper functions are scary, but we have them in the old test code, so :+1: